homeassistant/components/netatmo/netatmo_entity_base.py

```diff
@@ -58,13 +58,13 @@ async def async_added_to_hass(self) -> None:
                 )
 
             else:
-                await self.data_handler.register_data_class(
+                await self.data_handler.subscribe(
                     data_class["name"], signal_name, self.async_update_callback
                 )
 
-            for sub in self.data_handler.data_classes[signal_name].subscriptions:
+            for sub in self.data_handler.publisher[signal_name].subscriptions:
                 if sub is None:
-                    await self.data_handler.unregister_data_class(signal_name, None)
+                    await self.data_handler.unsubscribe(signal_name, None)
```

Comment:
Do you explicitly add a None into subscriptions some where? Is this really valid?

Reply:
Yes, but this is completely refactored in #73482.